EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockSP.cc

```diff
@@ -229,138 +229,205 @@ namespace l1t {
+		 Hit->Subsector() + 3 != conv_vals_SP.at(0) ) continue;
 	    
-	    if (St_hits[2] == 0 ) {
-	      Track_.push_Hit( res_hit->at(iHit) );
+	    if (St_hits.at(2) == 0 ) {
+	      Track_.push_Hit( (*Hit) );
 	      mu_.setTrackSubAddress( RegionalMuonCand::kME3Seg, SP_.ME3_stub_num() );
 	      mu_.setTrackSubAddress( RegionalMuonCand::kME3Ch,  
 				      L1TMuonEndCap::calc_uGMT_chamber( conv_vals_SP.at(0), 
 									conv_vals_SP.at(2), 
 									conv_vals_SP.at(3), 3) );
 	    }
-	    St_hits[2] += 1;
-	  } // End conditional: if ( (res_hit->at(iHit)).Station() == 3 
+	    St_hits.at(2) += 1;
+	  } // End conditional: if ( Hit->Station() == 3 
 	  
 	  
 	  // Match hit in station 4
 	  conv_vals_SP = convert_SP_location( SP_.ME4_CSC_ID(), (res->at(iOut)).PtrEventHeader()->Sector(), -99, 4 );
 	  
-	  if ( (res_hit->at(iHit)).Station()  == 4                  &&	    
-	       (res_hit->at(iHit)).Sector()   == conv_vals_SP.at(1) &&
-	       (res_hit->at(iHit)).Neighbor() == conv_vals_SP.at(3) &&
-	       (res_hit->at(iHit)).Stub_num() == SP_.ME4_stub_num() ) {
+	  if ( Hit->Station()  == 4                  &&	    
+	       Hit->Sector()   == conv_vals_SP.at(1) &&
+	       Hit->Neighbor() == conv_vals_SP.at(3) &&
+	       Hit->Stub_num() == SP_.ME4_stub_num() ) {
 	    
-	    if ( (res_hit->at(iHit)).Is_CSC() == 1 && 
-		 (res_hit->at(iHit)).CSC_ID() != conv_vals_SP.at(0) ) continue;
+	    if ( Hit->Is_CSC() == 1 && 
+		 Hit->CSC_ID() != conv_vals_SP.at(0) ) continue;
 	    
-	    if ( (res_hit->at(iHit)).Is_RPC() == 1 &&
-		 (res_hit->at(iHit)).Subsector() + 3 != conv_vals_SP.at(0) ) continue;
+	    if ( Hit->Is_RPC() == 1 &&
+		 Hit->Subsector() + 3 != conv_vals_SP.at(0) ) continue;
 	    
-	    if (St_hits[3] == 0 ) {
-	      Track_.push_Hit( res_hit->at(iHit) );
+	    if (St_hits.at(3) == 0 ) {
+	      Track_.push_Hit( (*Hit) );
 	      mu_.setTrackSubAddress( RegionalMuonCand::kME4Seg, SP_.ME4_stub_num() );
 	      mu_.setTrackSubAddress( RegionalMuonCand::kME4Ch,  
 				      L1TMuonEndCap::calc_uGMT_chamber( conv_vals_SP.at(0), 
 									conv_vals_SP.at(2), 
 									conv_vals_SP.at(3), 4) );
 	    }
-	    St_hits[3] += 1;
-	  } // End conditional: if ( (res_hit->at(iHit)).Station() == 4 
+	    St_hits.at(3) += 1;
+	  } // End conditional: if ( Hit->Station() == 4 
 	  
-	} // End loop: for (unsigned int iHit = 0; iHit < res_hit->size(); iHit++)
+	} // End loop: for (auto Hit = res_hit->begin(); Hit != res_hit->end(); ++Hit)
       
+
+	// Special configuration for single-stub tracks from ME1/1
+	if (Track_.Mode() == 1) {
+
+	  // Infer ME1/1 chamber based on track phi
+	  int chamber_min = ((Track_.GMT_phi() - 17) / 16) + Track_.Sector()*6 - 3;
+	  int chamber_max = ((Track_.GMT_phi() +  1) / 16) + Track_.Sector()*6 - 3;
+	  for (int iChamb = chamber_max; iChamb >= chamber_min; iChamb--) {
+	    int chamber = (iChamb < 37 ? iChamb : (iChamb % 36));
+
+	    for (auto Hit = res_hit->begin(); Hit != res_hit->end(); ++Hit) {
```

Comment:
hi @abrinke1 - could this and the loop below be range based loops (for ( auto const & Hit : res_hit)...)

Reply:
When I change to the range-based loop, I get the following compile error:
  * no matching function for call to 'begin(std::vector<l1t::EMTFHit>*&)'
If you have a suggestion for how to fix this, I can implement it.

Reply:
hi @abrinke1 , what syntax did you try ?
Just looking at the snippet, this
```
for (auto const & hit: * res_hit) {
```
should work.

Reply:
Fixed using suggestion from @fwyzard